operation/operation.c

```diff
@@ -245,28 +248,32 @@ void calibrate(oper_st * state) {
 // Puts LED on.
 // Loops infinitely until switch is pressed.
 // Puts LED off after switch press.
-void wait_until_sw_pressed(SW * sw_proceed, LED * led, oper_st * state) {
-    set_led_brightness(led, PWM_SOFT);
+void wait_until_sw_pressed(oper_st * state) {
+    set_led_brightness(&state->led, PWM_SOFT);
     logf_msg(WAITING_FOR_SW, state, 0);
-    while (!switch_pressed_debounced(sw_proceed)) {
+    while (!switch_pressed_debounced(&state->sw_proceed)) {
+        if (switch_pressed_debounced(&state->sw_log)) {
+            read_log_entry(MAX_ENTRIES);
+        }
         sleep_ms(50);
     }
     logf_msg(SW_PRESSED, state,
-             1, sw_proceed->board_index);
-    set_led_brightness(led, PWM_OFF);
+             1, state->sw_proceed.board_index);
+    set_led_brightness(&state->led, PWM_OFF);
 }
 
 // Dispense pills according 'current_comp_idx', which is read from EEPROM on boot.
-void dispense(oper_st * state, LED * led) {
-    SW sw_log = init_switch(SW_1); // to print log
-
+void dispense(oper_st * state) {
     logf_msg(DISPENSE_CONTINUED, state, 1, state->current_comp_idx);
 
     uint64_t start = TIME_S;
     uint8_t compartment_left = PILLCOMP_COUNT - state->current_comp_idx;
     for (uint8_t comp = 0; comp < compartment_left; comp++) {
         uint64_t next_pilling_time = start + PILL_INTERVAL_S * comp;
         while (TIME_S < next_pilling_time) {
+            if (switch_pressed_debounced(&state->sw_log)) {
+                read_log_entry(MAX_ENTRIES);
+            }
```

Comment:
since this code has been repeated in 3 places, my suggestion is to put it under 1 function and call it. 
if (switch_pressed_debounced(&state->sw_log)) {
                read_log_entry(MAX_ENTRIES);
            }
  Also instead of MAX_ENTRIES as parameters I would consider to use &state->eeprom_log_idx

Reply:
Sure.